datasets.py

```diff
@@ -31,47 +38,188 @@ def create_mnist_datasets(
     test_dataset = tf.data.Dataset.from_tensor_slices(test_images).shuffle(test_size).batch(batch_size)
     return train_dataset, test_dataset
 
+'''
+Spoken digit audio dataset
+'''
+
+def _preprocess_spoken_digit_audio_tf(example, with_labels):
+    audio = example['audio']
+    # convert from int16? to float32 [-1.0, 1.0]
+    if audio.dtype.is_integer:
+        # detected clipping using just tf.int16.max to normalize...
+        audio = tf.cast(audio, tf.float32) / (tf.int16.max + 1.0)
```

Comment:
Normalization should not depend on sample. If we truly need to mix data coming from different sources, such that they'd have different types and ranges, we'd have a lot more to worry about such as differences in distributions. For now, let's assume that all data has the same type and range. More on what to do in the comment below.

Reply:
The things I'm doing here are audio-specific and shouldn't be construed as the kind of data standardization you're referring to. While it's true that we may eventually want to use this code with data that's encoded in something other than 16-bit I was under the impression that, like the other functions for the image dataset, we're tailoring the pipeline to this particular set of files for now, which IS stored as 16-bit. 

This particular block is for converting audio data stored as integers to floating point [-1, 1], which is how we want to actually work with it. More below